### ISSUE TITLE
Literal unquoting operators

### ISSUE THREAD

**Issue by egnha** [tidyverse/rlang#264]
In certain situations, it'd be nice to be able to quote unquoting operators. For example, you might want to generate the body of a function (as a quosure) and selectively unquote certain parts while embedding literal unquoting operators in other parts.

In other words, it would be convenient if `quo()` and the like could comprehend quoted unquoting operators, say `QUQ()`, `QUQS()`, `QUQE()`.

I have something like this in mind:
```R
quq <- lapply(set_names(c("UQ", "UQS", "UQE")), as.name)
names(quq) <- paste0("Q", names(quq))

quo_ <- function(x) {
  quo <- enquo(x)
  expr <- f_rhs(quo)
  expr <- eval(substitute(substitute(., quq), list(. = expr)))
  new_quosure(expr, f_env(quo))
}
```
`quo_()` would behave exactly like `quo()`, but would allow literal unquoting operators to be expressed:
```R
x <- quote(fun)

quo_(x)
#> <quosure: global>
#> ~x

quo_(!! x)
#> <quosure: global>
#> ~fun

quo_(UQ(x)(QUQ(x)))
#> <quosure: global>
#> ~fun(UQ(x))
```
(The last example illustrates unquoting vs literal unquoting, which would be convenient for metaprogramming.)

**Comment by lionel-** [comment]
I think it's not worth it to complicate the API and implementation. You could do this instead:

```r
quo(UQ(x)(!! quote(UQ(x))))
```

**Comment by egnha** [comment]
Thanks for your suggestion. It's a bit verbose, but does the trick!

In any case, I see your point. Maybe adding extra verbs is not worth the trouble.

(Perhaps I should also add that the reason I wanted this is that I found that in writing anonymous function bodies with [`nofrills::fn()`](https://github.com/egnha/nofrills), it was useful to allow both unquoting and literal unquoting. For `nofrills::fn()`, I found a simple workaround, basically what I was suggesting here.)
